nautobot_device_lifecycle_mgmt/models.py

```diff
@@ -157,7 +288,12 @@ def get_for_object(self, obj):
 class ValidatedSoftwareLCM(PrimaryModel):
     """ValidatedSoftwareLCM model."""
 
-    software = models.ForeignKey(to="dcim.SoftwareVersion", on_delete=models.CASCADE, verbose_name="Software Version")
+    software = models.ForeignKey(
+        to="dcim.SoftwareVersion", on_delete=models.CASCADE, verbose_name="Software Version", null=True
+    )
+    old_software = models.ForeignKey(
+        to="SoftwareLCM", on_delete=models.CASCADE, verbose_name="Software Version", blank=True, null=True
```

Comment:
Should this be `on_delete=models.SET_NULL` instead?  I would think that cleaning up the SoftwareLCM models and having it delete a ValidatedSoftwareLCM record would be undesirable.

Reply:
```suggestion
        to="SoftwareLCM", on_delete=models.SET_NULL, verbose_name="Software Version", blank=True, null=True
```

Reply:
Agreed. This might end up surprising end users if they delete `SoftwareLCM` instance post-migration and a related Validated Software disappears.